clang/include/clang/AST/Type.h

```diff
@@ -520,7 +520,9 @@ class Qualifiers {
            // to implicitly cast into the default address space.
            (A == LangAS::Default &&
             (B == LangAS::cuda_constant || B == LangAS::cuda_device ||
-             B == LangAS::cuda_shared));
+             B == LangAS::cuda_shared)) ||
+           // TODO make following check specific to AIE target only
```

Comment:
It seems to say: any non-default address space is a subspace of the default address space. I like that, and I think it would make sense to get that option into a target plugin.

Reply:
Yes the plan is have a target plugin, we can discuss on IM more. 
But thats a big change in clang frontend and based on discussion with Gaetan & Konstantin, on side I work on that to get upstream to LLVM